From 512e24568e4570af41b31007a6f58e9664df80a4 Mon Sep 17 00:00:00 2001
From: Yu, Linda <linda.yu@intel.com>
Date: Fri, 27 Sep 2013 08:34:37 +0800
Subject: set audio to be EOS when video got eos

BZ: 140634

Before MEDIA_PLAYBACK_COMPLETE, we need to make sure audiio/video set to
EOS both.

Category: feature differentiation
Domain: Video.Media-VPP
Origin:  internal
Upstream-Candidate: No

Change-Id: Ia11e00f37150885cedcdd1b91efa7e1732388033
Orig-Change-Id: I3e44ba49a4ee3f62c3b4438201156096bda03a95
Signed-off-by: Yu, Linda <linda.yu@intel.com>
Reviewed-on: http://android.intel.com:8080/134466
Reviewed-by: Wang, Xigui <xigui.wang@intel.com>
Reviewed-by: Ding, Haitao <haitao.ding@intel.com>
Tested-by: Ding, Haitao <haitao.ding@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 media/libstagefright/AwesomePlayer.cpp |   16 ++++++++++++----
 1 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/media/libstagefright/AwesomePlayer.cpp b/media/libstagefright/AwesomePlayer.cpp
index 143ef45..3f56029 100644
--- a/media/libstagefright/AwesomePlayer.cpp
+++ b/media/libstagefright/AwesomePlayer.cpp
@@ -2002,6 +2002,8 @@ void AwesomePlayer::onVideoEvent() {
     MediaBuffer *tmpVideoBuffer = mVideoBuffer;
     mVideoBuffer = NULL;
 
+    SeekType seekTmpWithVpp = mSeeking;
+
     if(mVPPProcessor->canSetDecoderBufferToVPP()) {
     if (!mVideoBuffer) {
         MediaSource::ReadOptions options;
@@ -2050,10 +2052,10 @@ void AwesomePlayer::onVideoEvent() {
                     startAudioPlayer_l();
                 }
 
-                if (seekTemp != NO_SEEK) {
-                    modifyFlags(AUDIO_AT_EOS, SET); // video is eos, end the audio?
-                    mVideoTimeUs = mSeekTimeUs;
-                }
+                //if (seekTemp != NO_SEEK) {
+                //    modifyFlags(AUDIO_AT_EOS, SET); // video is eos, end the audio?
+                //    mVideoTimeUs = mSeekTimeUs;
+                //}
                 mVPPProcessor->setEOS();
 
                 //modifyFlags(VIDEO_AT_EOS, SET);
@@ -2102,6 +2104,12 @@ void AwesomePlayer::onVideoEvent() {
         if (err == ERROR_END_OF_STREAM) {
             ALOGV("VPP finished");
             CHECK(mVideoBuffer == NULL);
+
+            if (seekTmpWithVpp != NO_SEEK) {
+                modifyFlags(AUDIO_AT_EOS, SET); // video is eos, end the audio?
+                mVideoTimeUs = mSeekTimeUs;
+            }
+
             modifyFlags(VIDEO_AT_EOS, SET);
             postStreamDoneEvent_l(err);
             return;
-- 
1.7.4.1

